packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,642 @@
+    const radixPrivateKey = this.getRadixPrivateKey()
+
+    if (params.asset == undefined) {
+      params.asset = XrdAsset
+    }
+
+    const preparedTransaction = await this.prepareTx(params)
+    const compiledTransactionManifest = await Convert.HexString.toUint8Array(preparedTransaction.rawUnsignedTx)
+    const transactionManifest = await RadixEngineToolkit.TransactionManifest.decompile(
+      compiledTransactionManifest,
+      networkId,
+    )
+    try {
+      const gatewayStatusResponse: GatewayStatusResponse = await this.gatewayApiClient.status.getCurrent()
+      const notarizedTransaction = await TransactionBuilder.new().then((builder) =>
+        builder
+          .header({
+            networkId: networkId,
+            startEpochInclusive: gatewayStatusResponse.ledger_state.epoch,
+            endEpochExclusive: gatewayStatusResponse.ledger_state.epoch + 10,
+            nonce: generateRandomNonce(),
+            notaryPublicKey: radixPrivateKey.publicKey(),
+            notaryIsSignatory: true,
+            tipPercentage: 0,
+          })
+          .plainTextMessage(params.memo ? params.memo : '')
+          .manifest(transactionManifest)
+          .notarize(radixPrivateKey),
+      )
+
+      const compiledTransaction = await RadixEngineToolkit.NotarizedTransaction.compile(notarizedTransaction)
+      const compiledTransactionHex = Convert.Uint8Array.toHexString(compiledTransaction)
+
+      if (broadcastTx) {
+        await this.broadcastTx(compiledTransactionHex)
+      }
+      const transactionId = await RadixEngineToolkit.NotarizedTransaction.intentHash(notarizedTransaction)
+      return transactionId.id
+    } catch (error) {
+      throw new Error('Failed to transfer')
+    }
+  }
+  /**
+   * Submits a transaction
+   * @param txHex - The transaction hex build with the transfer method
+   * @returns - The response from the gateway
+   */
+  async broadcastTx(txHex: string): Promise<string> {
+    await this.gatewayApiClient.transaction.innerClient.transactionSubmit({
+      transactionSubmitRequest: {
+        notarized_transaction_hex: txHex,
+      },
+    })
+    const notarizedTransaction = await RadixEngineToolkit.NotarizedTransaction.decompile(
+      Convert.HexString.toUint8Array(txHex),
+    )
+    const intentHash = await RadixEngineToolkit.NotarizedTransaction.intentHash(notarizedTransaction)
+    return Convert.Uint8Array.toHexString(intentHash.hash)
+  }
+
+  /**
+   * Prepares a transaction to be used by the transfer method
+   * It will include a non signed transaction
+   * @param params - The transaction params
+   * @returns a PreparedTx
+   */
+  async prepareTx(params: TxParams): Promise<PreparedTx> {
+    if (params.asset == undefined) {
+      params.asset = XrdAsset
+    }
+    const fromAddress = await this.getAddressAsync()
+    const stringManifest = `
+    CALL_METHOD
+      Address("${fromAddress}")
+      "lock_fee"
+      Decimal("${this.feeBounds.upper}");
```

Comment:
I do not think that we should lock the upper bound, I think that we should use the estimated fees.

Reply:
fixed in the refactor